perses/app/example.py

```diff
@@ -1,3 +1,670 @@
+import time
+import yaml
+import networkx as nx
+
+# environments
+"""
+- ligand-protein complex (includes small-molecule ligands and peptides)
+- ligand-water
+- ligand-vacuum
+- ligand-membrane
+- ligand-amorphous solid
+
+*for clarity's sake, 'ligand' is the object containing a residue that will be transformed alchemically
+"""
+
+from openmmforcefields.generators import SystemGenerator
+system_generator = SystemGenerator(forcefields=["protein.ff14SB.xml", "tip3p_standard.xml"],small_molecule_forcefield="gaff-2.11")
+
+# Define the thermodynamic state we will be simulating in
+from openmmtools.states import ThermodynamicState
+
+thermodynamic_state = ThermodynamicState(
+    temperature=300.0 * unit.kelvin, pressure=1.0 * unit.atmospheres, pH=7.0
+)
+
+# Load receptor and ligand using the openforcefield topology objects
+from openforcefield.topology import Molecule
+
+# will need to add biopolymer support, but can be done quickly
+receptor = Molecule.from_file("receptor.pdb")
+# multiple ligands pre-docked to site
+ligands = Molecule.from_file("docked_ligands.sdf")
+
+
+"""
+CASE 1 : Relative Binding Free Energies to a protein target
+"""
+receptor = Molecule.from_file("receptor.pdb")
+complex_environment = SmallMoleculeComplexEnvironment(thermodynamic_state, receptor=receptor)
+aqueous_environment = AqueousEnvironment(thermodynamic_state)
+aqueous_environment.solvent_buffer = 9 * unit.angstroms
+octanol_environment = OctanolEnvironment(thermodynamic_state)
+#amorphous_solid_environment = AmorphousSolidEnvironment(thermodynamic_state)
+
+#presumably there are specialized arguments for each Environment that is supported
+
+receptor2 =  Molecule.from_file("receptor2.pdb") #maybe we want to add another phase with a similar protein that could have either some fixed mutations or spectators
+spectator_ligands = ligands = Molecule.from_file("spectator_ligands.sdf")
+complex2_environment = SmallMoleculeComplexEnvironment(thermodynamic_state, receptor = receptor2, spectators = spectator_ligands)
+
+#can pass attributes like above OR pass a yaml into the `Environment` object __init__
+
+
+
+from perses.network import FreeEnergyNetwork
+#add pre-specified topology_proposal, hybrid_factory, geometry_engine, and sampler arguments from a yaml
+yaml_file = open('setup.yaml', 'r')
+setup_options = yaml.load(yaml_file, Loader=yaml.FullLoader)
+yaml_file.close()
+
+network = FreeEnergyNetwork(system_generator,
+                            ligands, #list of ligands (to which more ligands can be appended)
+                            transformation_specifications = setup_options, #a yaml dict that has prespecified parameters for proposals and/or samplers that overwrite defaults
+                            environments = [complex_environment, aqueous_environment, octanol_environment, complex2_environment] #a list of environments
+                            )
```

Comment:
there needs to be a `factory` object that actually builds the `FreeEnergyNetwork`.